SECURITY.md

```diff
@@ -27,7 +33,7 @@ The SRC members will share various tasks as listed below:
 
 ### Contacting the Security Response Committee
 
-Contact the team by sending email to the [SRC Members](#SRC-Members)
```

Comment:
Do we have [security@sigstore.dev](mailto:security@sigstore.dev) set up and how will we handle encrypted mail?

Reply:
I've updated with an instruction to contact the email list if they want to use GPG. If we see multiple requests, we can create a shared static key. 

Otherwise, I'd like to avoid the reporter having to encrypt with multiple GPG keys.